### PR TITLE
Templates and Themes core functionalities

### PR DESCRIPTION
**Pull request by sadanandpai**
- Added templates into the header with selection functionality
- Added themes into the header with selection functionality
- Added keys in missing places for loops using the map
(UI is not developed. Dummy UI is used)

### CODE REVIEW

File: src/components/core/settings-header/TemplatesSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from "react";
+import { Popover, Button } from "antd";
+import { useTemplates } from "../../stores/settings.store";
+
+export function TemplateSettings() {
+  const setTemplate = useTemplates((state: any) => state.setTemplate);
+
+  const content = (
+    <div>
+      <Button onClick={() => setTemplate(0)}>0</Button>
```
Comment (gopal1996) [inline src/components/core/settings-header/TemplatesSettings.tsx:10]:
```suggestion
      <Button onClick={() => useTemplates.getState().setTemplate(0)}>0</Button>
```

Comment (gopal1996) [reply src/components/core/settings-header/TemplatesSettings.tsx:10]: setTemplate is a function . so when we initialize like const setTemplate = useTemplates((state: any) => state.setTemplate);. zustand add a listner. but for function call, we can directly call in non-reactive way

**Review by gopal1996 — CHANGES_REQUESTED** [review]

File: src/components/core/settings-header/TemplatesSettings.tsx (hunk 1 above)
Comment (sadanandpai) [reply src/components/core/settings-header/TemplatesSettings.tsx:10]: image
They have recommended like this in React components in docs. Confused which is the best approach

**Review by gopal1996 — APPROVED** [review]

File: src/components/core/settings-header/TemplatesSettings.tsx (hunk 1 above)
Comment (gopal1996) [reply src/components/core/settings-header/TemplatesSettings.tsx:10]: yes. we can do this also. we will explore this part later

**Pull request merged**